app/uk/gov/hmrc/tariffclassificationfrontend/controllers/SampleController.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tariffclassificationfrontend.controllers
+
+import javax.inject.{Inject, Singleton}
+import play.api.data.Form
+import play.api.i18n.MessagesApi
+import play.api.mvc.{Action, AnyContent, Call, Request}
+import play.twirl.api.Html
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.tariffclassificationfrontend.config.AppConfig
+import uk.gov.hmrc.tariffclassificationfrontend.forms.SampleStatusForm
+import uk.gov.hmrc.tariffclassificationfrontend.models.{Case, Operator, Permission}
+import uk.gov.hmrc.tariffclassificationfrontend.models.SampleStatus.SampleStatus
+import uk.gov.hmrc.tariffclassificationfrontend.models.request.AuthenticatedRequest
+import uk.gov.hmrc.tariffclassificationfrontend.service.CasesService
+import uk.gov.hmrc.tariffclassificationfrontend.views
+import uk.gov.hmrc.tariffclassificationfrontend.views.CaseDetailPage
+
+import scala.concurrent.Future
+import scala.concurrent.Future.successful
+
+@Singleton
+class SampleController @Inject()(override val verify: RequestActions,
+                                  override val caseService: CasesService,
+                                  override val messagesApi: MessagesApi,
+                                  override implicit val config: AppConfig) extends StatusChangeAction[Option[SampleStatus]] {
+
+  override protected val requiredPermission: Permission.Value = Permission.EDIT_SAMPLE
+
+  override protected def redirect: String => Call = routes.CaseController.trader
+
+  override protected def isValidCase(c: Case)(implicit request: AuthenticatedRequest[_]): Boolean = {
+    true //No constraints on when the case is valid
```

Comment:
Do we need to double check this statement?  E.g. should it be possible to change sample status when case is complete?

Reply:
From the amigos sessions it should be available to do at any stage of case lifecycle, as sometimes officers will complete the case, then mark the sample as returned later on for instance

Reply:
Fair enough